inline_img_protocol_autodetect.c

```diff
@@ -0,0 +1,127 @@
+/* The only externally visible symbol exported by this module is
+ * inline_img_protocol_autodetect(). I appreciate that this name does not
+ * roll off the tongue
```

Comment:
Are you trolling?

Reply:
No? I certainly intended no offence, I must have just been in a good mood while writing that, sorry